tests/difficulty/difficulty.cpp

```diff
@@ -64,7 +64,7 @@ int main(int argc, char *argv[]) {
         }
         uint64_t res = cryptonote::next_difficulty(
             vector<uint64_t>(timestamps.begin() + begin, timestamps.begin() + end),
-            vector<uint64_t>(cumulative_difficulties.begin() + begin, cumulative_difficulties.begin() + end), DEFAULT_TEST_DIFFICULTY_TARGET);
+            vector<uint64_t>(cumulative_difficulties.begin() + begin, cumulative_difficulties.begin() + end), DEFAULT_TEST_DIFFICULTY_TARGET, 1);
```

Comment:
n + 1 ? Guess it doesn't matter as long as < threshold though

Reply:
I'd rather have it this way, since the height is used to conditionally make an adjustment to the output diff value; see the end of `next_difficulty()`:

```c++
    difficulty_type new_diff = (low + time_span - 1) / time_span;
    if (height >= HARDFORK_1_HEIGHT && height < HARDFORK_1_HEIGHT+HARDFORK_1_DIFFADJ_WINDOW) {
      new_diff += new_diff*(HARDFORK_1_HEIGHT+HARDFORK_1_DIFFADJ_WINDOW-height)*(HARDFORK_1_DIFFADJ-1)/HARDFORK_1_DIFFADJ_WINDOW;
    }
    return new_diff;
```

It doesn't matter because the maximum height stored in tests/difficulty/data.txt is only 29000, but the test may fail if someone added some bad numbers to data.txt. I'd rather have the test completely ignore this adjustment by always setting the height to 1.

Reply:
> because the maximum height stored in tests/difficulty/data.txt is only 29000

I was wrong here, 29000 represents a timestamp. The height corresponds to the number of lines in data.txt, which is 1000. Anyway, the reason of not using n+1 as the height parameter to `next_difficulty` remains the same.